src/components/Animal/Animal.spec.js

```diff
@@ -0,0 +1,116 @@
+import Animal from './Animal';
+import '@testing-library/jest-dom';
+import { render, screen } from '@testing-library/react';
+
+describe('Single Animal-Display', () => {
+	it("renders every important detail of animal & the names of spots where animal lives & NOT the names of spots where animal is not living  & the names of organizers who arrange diving with animal & NOT the names of organiZers who don't arrange diving with animal ", () => {
+		render(
+			<Animal
+				spots={[
+					{
+						id: '1',
+						name: 'Bimini',
+						country: 'Bahamas',
+						geoTag: { latitude: '25.703188', longitude: '-79.281459' },
+						description: 'nice and warm',
+					},
+					{
+						id: '2',
+						name: 'Gran Bahama',
+						country: 'Bahamas',
+						geoTag: { latitude: '26.661393', longitude: '-78.268353' },
+						description: 'nice and warm',
+					},
+					{
+						id: '3',
+						name: 'Galápagos',
+						country: 'Ecuador',
+						geoTag: { latitude: '-0.628815', longitude: '-90.363875' },
+						description: 'nice and warm',
+					},
+					{
+						id: '4',
+						name: 'Socorro Islands',
+						country: 'Mexico',
+						geoTag: { latitude: '18.792742', longitude: '-110.982031' },
+						description: 'nice and warm',
+					},
+				]}
+				animal={{
+					id: '1',
+					name: 'Dolphin',
+					spots: ['Gran Bahama', 'Bimini', 'Galápagos'],
+					description: 'friendly, very intelligent, likes to play',
+				}}
+				organizers={[
+					{
+						id: '1',
+						name: 'Unexso',
+						spots: [
+							{
+								name: 'Freeport',
+								area: 'Gran Bahama',
+								animals: ['Dolphin'],
+							},
+						],
+					},
+					{
+						id: '2',
+						name: 'Silversea Cruises',
+						spots: [
+							{
+								name: 'San Cristóbal',
+								area: 'Galápagos',
+								animals: ['Dolphin', 'Shark'],
+							},
+							{
+								name: 'Socorro Islands',
+								area: 'Colima, Revillagigedo Archipelago',
+								animals: ['Shark', 'Dolphin'],
+							},
+						],
+					},
+					{
+						id: '3',
+						name: 'Pacific Fleet',
+						spots: [
+							{
+								name: 'Neptun Islands',
+								area: 'South Australia, Spencer Gulf',
+								animals: ['Shark'],
+							},
+						],
+					},
+				]}
+			/>
+		);
+
+		const nameSpot1 = screen.queryByText(/Bimini/i);
+		const nameSpot2 = screen.queryByText(/Gran Bahama/i);
+		const nameSpot3 = screen.queryByText(/Socorro Islands/i);
+
+		const animalDetail1 = screen.queryByText(/Dolphin/i);
```

Comment:
I think the names are rendered as `h2`, right? You can thus use `getByRole` with a heading role in order to be more precise what you are looking for.